### PR TITLE
Implement createDirectory for apple platform.

### LINKED ISSUE #15073: FileUtils::createDirectory fails on OS X app with sandbox
If you want to release application to Mac App Store, you must turn on Sanbox entitlement.
But **createDirectory** function fails to work with it, because function **mkdir** returns _-1_, not directory exists flag, for folders, that are not in Sandbox.( if pass full path to directory in function createDirectory).

``` c++
for (int i = 0; i < dirs.size(); ++i)
    {
        subpath += dirs[i];
        dir = opendir(subpath.c_str());

        if (!dir)
        {
            // directory doesn't exist, should create a new one

            int ret = mkdir(subpath.c_str(), S_IRWXU | S_IRWXG | S_IRWXO);
            if (ret != 0 && (errno != EEXIST))
            {
                // current directory can not be created, sub directories can not be created too
                // should return
                return false; // <------ Here function exits on '/' folder if path if absolute
            }
        }
        else
        {
            // directory exists, should close opened dir
            closedir(dir);
        }
    }
```

### PR DESCRIPTION
**Pull request by 1scaR1**
Fixes #15073.
I think using of file manger for creating directories is more suitable for apple platforms.

### CODE REVIEW

File: cocos/platform/apple/CCFileUtils-apple.mm [hunk 1]
```diff
@@ -534,4 +534,16 @@ static int unlink_cb(const char *fpath, const struct stat *sb, int typeflag, str
...
+{
+    CCASSERT(!path.empty(), "Invalid path");
+    
+    if (isDirectoryExist(path))
+        return true;
+    
+    NSError* error;
+    
+    return [s_fileManager createDirectoryAtPath:[NSString stringWithUTF8String:path.c_str()] withIntermediateDirectories:YES attributes:nil error:&error];
+}
```
Comment (minggo) [inline cocos/platform/apple/CCFileUtils-apple.mm:554]: I think we should remove error if it is not used, or we should log the error message.

**Comment by 1scaR1** [comment]
@minggo Added error logging.
Testing can be simple: turn on `Sandbox entitlement` ant try to create directory such as `/abc`

**Comment by minggo** [comment]
@1scaR1 I applied your PR, and it also can not create a directory. My code is:

``` c++
bool result = FileUtils::getInstance()->createDirectory("/abc");
if (result)
    CCLOG("true");
```

**Comment by 1scaR1** [comment]
And it must not. In sandbox you can create dirs only in writable path. 
But if there is no sandbox you do not have rights in root dir.
This fix allows to create dir such as `getWritablePath()+"abc"` or `fullPathForXXX("dir")` with sandbox enabled.

**Comment by minggo** [comment]
@1scaR1 Without this PR, i can create directory too, my codes changed to

``` c++
auto fileUtils = FileUtils::getInstance();
    bool result = fileUtils->createDirectory(fileUtils->getWritablePath() + "/abc");
    if (result)
        CCLOG("true");
```

And i already enable sandbox.

**Comment by minggo** [comment]
The configuration is:

screen shot 2016-04-19 at 5 36 11 pm

**Comment by minggo** [comment]
I tested in simulator, should i test in devices?

**Comment by 1scaR1** [comment]
@minggo I have your code(without this PR) above not logging `true`(with sandbox enabled, **directory must not exists**).
Also i printed writable path when directory not creates, ant it's like:
`/Users/_USER_/Library/Containers/_APP_ID_/Data/Documents/`

It not works only on OS X, why there:

> I tested in simulator, should i test in devices?

?

**Comment by 1scaR1** [comment]
XCode 7.3, OS X 10.11.4(and same effect on 10.11.5 beta 1)

**Comment by minggo** [comment]
> I tested in simulator, should i test in devices?

Oh, yep. It's my fault. I tested on Mac. I will double check it.

**Comment by minggo** [comment]
@1scaR1 You are right. I think i didn't remove `abc` folder when testing without this PR.
Thanks.

PS: You publish games to Mac Store?

**Comment by 1scaR1** [comment]
@minggo , yes I published my game)

**Comment by minggo** [comment]
Congratulations.

**Comment by 1scaR1** [comment]
Thanks)

**Pull request merged**